.github/scripts/userdata.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+set -e
+REPO_ROOT=`git rev-parse --show-toplevel`
+ES_VER=`$REPO_ROOT/bin/version-info --es`
+ODFE_VER=`$REPO_ROOT/bin/version-info --od`
+echo $ES_VER $ODFE_VER
+
+if [ "$#" -eq 0 ] || [ "$#" -gt 2 ] || [ -z "$1" ] || [ -z "$2" ]
+then
+    echo "Please assign 2 parameters when running this script"
+    echo "Format for dispatch event: \"client_payload\": {
+		\"distribution\" : \"rpm\",
+		\"security\" : \"enable\"
+	}"
+    echo "Example: $0 \"RPM\" \"ENABLE\""
+    echo "Example: $0 \"DEB\" \"DISABLE\""
+    exit 1
+fi
+
+###### RPM package with Security enabled ######
+if [ "$1" = "RPM" ]
+then
+cat <<- EOF > $REPO_ROOT/userdata_$1.sh
+#!/bin/bash
+sudo -i
+sudo curl https://d3g5vo6xdbdb9a.cloudfront.net/yum/staging-opendistroforelasticsearch-artifacts.repo -o /etc/yum.repos.d/staging-opendistroforelasticsearch-artifacts.repo
+sudo yum install -y opendistroforelasticsearch-$ODFE_VER
+sudo sysctl -w vm.max_map_count=262144
+echo "node.name: init-master" >> /etc/elasticsearch/elasticsearch.yml
+echo "cluster.name: odfe-$ODFE_VER-rpm-auth" >> /etc/elasticsearch/elasticsearch.yml
+echo "network.host: 0.0.0.0" >> /etc/elasticsearch/elasticsearch.yml
+echo "cluster.initial_master_nodes: [\"init-master\"]" >> /etc/elasticsearch/elasticsearch.yml
+
+# Start the service
+sudo systemctl start elasticsearch.service
+sleep 30
+
+# Installing kibana
+sudo yum install -y opendistroforelasticsearch-kibana-$ODFE_VER
+echo "server.host: 0.0.0.0" >> /etc/kibana/kibana.yml
+EOF
+fi
+
+###### DEB package with Security enabled ######
+if [ "$1" = "DEB" ]
```

Comment:
I would suggest make every user input/param like this to be lower case only.
And add a check to always convert A-Z to a-z.

Reply:
Doing it [here](https://github.com/opendistro-for-elasticsearch/opendistro-build/pull/282/files#diff-f7892f93787763ebe62dac7c4727487aR32-R33)

Reply:
I dont really like this approach as it is relying on the workflow to check the inputs.
I would suggest let the script do it as the script is the one needs to process these inputs, not the workflow.

Reply:
And just on a personal note I do like to handle input in lower case better than upper case.
As that is more natural to most users.

Reply:
User may put any input the workflow is converting it to upper case

Reply:
Yeah, Upper Case is just a personal preference.
I do think the conversion needs to happen within the script, not outside.

Reply:
> I dont really like this approach as it is relying on the workflow to check the inputs.
> I would suggest let the script do it as the script is the one needs to process these inputs, not the workflow.

Cannot put in the script as those values are also used by cloudformation template to set up cluster according to the needs

Reply:
After discussion, Sayali will put the checks in scripts later.
Thanks.